### PR TITLE
batchnorm is clonable by adding the running estimates to constructor

### PR DESCRIPTION
**Pull request by soumith**
fixing batchnorm tests

fixes https://github.com/torch/nn/issues/282

### CODE REVIEW

**Comment by soumith** [comment]
A small PR to make BatchNorm be shareable. Tested thoroughly.

**Comment by Atcold** [comment]
Thank you @soumith.

**Comment by Atcold** [comment]
```lua
th> require 'nn';
                                                                      [0.0218s]
th> BN = nn.BatchNormalization(0)
/usr/local/share/lua/5.1/nn/BatchNormalization.lua:36: To set affine=false call BatchNormalization(nOutput,  eps, momentum, false) 
stack traceback:
        [C]: in function 'assert'
        /usr/local/share/lua/5.1/nn/BatchNormalization.lua:36: in function '__init'
        /usr/local/share/lua/5.1/torch/init.lua:51: in function </usr/local/share/lua/5.1/torch/init.lua:47>
        [C]: in function 'BatchNormalization'
        [string "BN = nn.BatchNormalization(0)"]:1: in main chunk
        [C]: in function 'xpcall'
        /usr/local/share/lua/5.1/trepl/init.lua:648: in function 'repl'
        /usr/local/lib/luarocks/rocks/trepl/scm-1/bin/th:185: in main chunk
        [C]: at 0x010946ea90
                                                                      [0.0035s]
th> 
```

**Comment by soumith** [comment]
yes, the constructor had to be changed for this to be shareable. I could not do lazy-initialization at runtime anymore.
